rust/src/delta.rs

```diff
@@ -1509,7 +1509,10 @@ fn process_action(state: &mut DeltaTableState, action: Action) -> Result<(), App
             state.files.push(v);
         }
         Action::remove(v) => {
-            state.files.retain(|a| *a.path != v.path);
+            let index = { state.files.iter().position(|a| *a.path == v.path) };
+            if let Some(index) = index {
+                state.files.remove(index);
```

Comment:
As the order is not important (I believe) `swap_remove` could be used, which is `O(1)` instead of `O(n)`

Reply:
I also think that order does not matter (although I am new to this repo). I am pushing the suggested change and the required changes in test. In test, I am sorting the vectors before comparing.

Reply:
That's a nice optimization, good call @Dandandan :)